### PR TITLE
update docker service related model

### PR DESCRIPTION
**Pull request by fengxx**
change command from String to List<String>, see also [spec](https://github.com/moby/moby/blob/master/api/types/swarm/container.go#L50)
add labels and forceUpdate

---
This change is [Reviewable](https://reviewable.io/reviews/docker-java/docker-java/917)

### CODE REVIEW

File: src/test/java/com/github/dockerjava/core/command/CreateServiceCmdExecTest.java [hunk 1]
```diff
@@ -59,11 +64,13 @@ public void testCreateService() throws DockerException {
                 .withListenAddr("127.0.0.1")
                 .withAdvertiseAddr("127.0.0.1")
                 .exec();
-
+        List<String> command = Arrays.asList("sleep", "300");
+        Map<String,String> labels=new HashMap<>();
+        labels.put("com.docker.java.usage","test");
         dockerClient.createServiceCmd(new ServiceSpec()
-                .withName(SERVICE_NAME)
+                .withName(SERVICE_NAME).withLabels(labels)
```
Comment (KostyaSha) [inline src/test/java/com/github/dockerjava/core/command/CreateServiceCmdExecTest.java]: please call fluent from new line

**Comment by KostyaSha** [comment]
could i postpone it until i resolve tests refactoring (atm fixing parallel execution issues)?

**Comment by KostyaSha** [comment]
Btw, are you testing master code?

File: src/test/java/com/github/dockerjava/core/command/CreateServiceCmdExecTest.java (hunk 1 above)
Comment (fengxx) [reply src/test/java/com/github/dockerjava/core/command/CreateServiceCmdExecTest.java]: fixed

**Comment by fengxx** [comment]
yes, I am testing on master code, please take your time for refactoring

**Comment by KostyaSha** [comment]
please rebase

**Comment by fengxx** [comment]
got `java.lang.Exception: Method beforeTest() should be static` when running in IDE, but `failsafe:integration-test` is fine
I am not sure whether it is ok to set a default value, e.g. null EndpointSpec.mode will be created as VIP,  set default value to VIP can help writing tests using `IsEqual` to compare inspect results

**Comment by KostyaSha** [comment]
m... i didn't look at swarm tests after migration. What IDE do you use?

**Comment by fengxx** [comment]
IntelliJ IDEA
```
java.lang.Exception: Method beforeTest() should be static

	at org.junit.runners.model.FrameworkMethod.validatePublicVoid(FrameworkMethod.java:93)
	at org.junit.runners.model.FrameworkMethod.validatePublicVoidNoArg(FrameworkMethod.java:74)
	at org.junit.runners.ParentRunner.validatePublicVoidNoArgMethods(ParentRunner.java:155)
	at org.junit.runners.ParentRunner.collectInitializationErrors(ParentRunner.java:125)
	at org.junit.runners.ParentRunner.validate(ParentRunner.java:416)
	at org.junit.runners.ParentRunner.<init>(ParentRunner.java:84)
	at org.junit.runners.Suite.<init>(Suite.java:112)
	at org.junit.runners.Parameterized.<init>(Parameterized.java:243)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
```

File: src/main/java/com/github/dockerjava/api/model/TaskSpec.java [hunk 2]
```diff
@@ -47,6 +49,14 @@
     private Driver logDriver;
 
     /**
+     * @since 1.24
+     * a counter that triggers an update even if no relevant parameters have been changed
+     * a random value will work if it is incrementing.
+     */
+    @JsonProperty("ForceUpdate")
+    private Integer forceUpdate = 0;
```
Comment (KostyaSha) [inline src/main/java/com/github/dockerjava/api/model/TaskSpec.java]: usually we don't define default values and using includeNonNull

**Comment by KostyaSha** [comment]
@fengxx feel free to change it, i only converted tests from testng, but they were failing even before. Not sure how to use them in travis.

**Comment by KostyaSha** [comment]
i thought bit more about swarm mode tests. Probably swarm init/etc should be done with DIND image and all other operations against just swarm endpoint, WDYT?

**Comment by sebastianbinder** [comment]
Any update on this?

**Comment by fengxx** [comment]
Rebase on master

**Comment by sebastianbinder** [comment]
Thanks for your work @fengxx and @KostyaSha! Really appreciate it being able to use services now. I just opened #958 regarding some services issue. Maybe one of you guys has an idea.

**Pull request merged**